### ISSUE TITLE
WSL2 ARM won't load compressed kernel

### ISSUE THREAD

**Issue by elsaco** [microsoft/WSL#9791]
### Windows Version

Microsoft Windows [Vesion 10.0.22621.1413]

### WSL Version

1.1.5.0

### Are you using WSL 1 or WSL 2?

- [X] WSL 2
- [ ] WSL 1

### Kernel Version

5.15.90.1

### Distro Version

Ubuntu 23.04

### Repro Steps

1. Build `linux-msft-wsl-5.15.90.1` using `config-wsl-arm64`
2. Add custom kernel setting to `.wslconfig`
3. Run `wsl --shutdown`
4. Start a distro

### Expected Behavior

WSL distro starts with custom kernel.

### Actual Behavior

If kernel is compressed WSL won't start. After minutes it shows timeout errors:

```
The operation timed out because a response was not received from the virtual machine or container.
Error code: Wsl/Service/CreateInstance/CreateVm/HCS_E_CONNECTION_TIMEOUT
Press any key to continue...

```

### Diagnostic Logs

It works if the custom kernel image is uncompressed.

**Comment by benhillis** [comment]
Repro steps are incomplete, which kernel binary are you using? For ARM64 we're booting from UEFI so it will need to contain the loader.

**Comment by elsaco** [comment]
These are the targets from `arch/arm64/boot/Makefile`:

`targets := Image Image.bz2 Image.gz Image.lz4 Image.lzma Image.lzo`

It defaults to `Image.gz` so running `make` builds a compressed kernel image. When added to `.wslconfig` it won't boot. If the kernel image is being uncompressed then WSL starts okay. Removing the extension `.gz` it won't work either. I tried `bz2` image also w/out success. There is no option to build `bzImage` for `arm64`

Tested on DevKit 2023 (a.k.a Volterra) but it shouldn't matter the platform.
